File: javav2/example_code/lambda/src/main/java/com/example/lambda/demo/LambdaInvoke.java
/**
 * Copyright 2010-2019 Amazon.com, Inc. or its affiliates. All Rights Reserved.
 *
 * This file is licensed under the Apache License, Version 2.0 (the "License").
 * You may not use this file except in compliance with the License. A copy of
 * the License is located at
 *
 * http://aws.amazon.com/apache2.0/
 *
 * This file is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR
 * CONDITIONS OF ANY KIND, either express or implied. See the License for the
 * specific language governing permissions and limitations under the License.
 *
 */

// snippet-comment:[These are tags for the AWS doc team's sample catalog. Do not remove.]
// snippet-sourcedescription:[DynamoDBAsyncGetItem.java demonstrates how to get an item by using the DynamoDbAsyncClient object]
// snippet-service:[dynamodb]
// snippet-keyword:[Java]
// snippet-keyword:[Amazon DynamoDB]
// snippet-keyword:[Code Sample]
// snippet-sourcetype:[full-example]
// snippet-sourcedate:[2019-11-19]
// snippet-sourceauthor:[AWS]

package com.example.lambda.demo;

import software.amazon.awssdk.services.lambda.LambdaClient;
import software.amazon.awssdk.regions.Region;
import software.amazon.awssdk.services.lambda.model.InvokeRequest;
import software.amazon.awssdk.core.SdkBytes;
import software.amazon.awssdk.services.lambda.model.InvokeResponse;


public class LambdaInvoke {

    public static void main(String[] args) {

        InvokeResponse res = null ;
        try
        {
            Region region = Region.US_WEST_2;
            LambdaClient awsLambda = LambdaClient.builder().region(region).build();

            //Need a SdkBytes instance for the payload
            SdkBytes payload = SdkBytes.fromUtf8String("{\n" +
                    " \"Hello \": \"Paris\",\n" +
                    " \"countryCode\": \"FR\"\n" +
                    "}" ) ;

            //Setup an InvokeRequest
            InvokeRequest request =  InvokeRequest.builder()
                    .functionName("arn:aws:lambda:us-west-2:335446330391:function:HelloFunction")
                    .payload(payload)
                    .build();

            //Invoke the Lambda function
            res = awsLambda.invoke(request);

            //Get the response
            String value = res.payload().asUtf8String() ;

            //write out the response
            System.out.println(value);

        }
        catch(Exception e)
        {
            e.getStackTrace();
        }
    }
}

File: .kotlin_alpha/services/lambda/src/main/kotlin/com/kotlin/lambda/LambdaInvoke.kt
// snippet-sourcedescription:[LambdaInvoke.kt demonstrates how to invoke a Lambda function by using the LambdaClient object.]
//snippet-keyword:[AWS SDK for Kotlin]
// snippet-keyword:[AWS Lambda]
// snippet-keyword:[Code Sample]
// snippet-sourcetype:[full-example]
// snippet-sourcedate:[11/04/2021]
// snippet-sourceauthor:[AWS-scmacdon]

/*
   Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
   SPDX-License-Identifier: Apache-2.0
*/

package com.kotlin.lambda

// snippet-start:[lambda.kotlin.invoke.import]
import aws.sdk.kotlin.services.lambda.LambdaClient
import aws.sdk.kotlin.services.lambda.model.InvokeRequest
import aws.sdk.kotlin.services.lambda.model.LambdaException
import kotlin.system.exitProcess
// snippet-end:[lambda.kotlin.invoke.import]

/**
To run this Kotlin code example, ensure that you have setup your development environment,
including your credentials.

For information, see this documentation topic:
https://docs.aws.amazon.com/sdk-for-kotlin/latest/developer-guide/setup.html
 */

suspend fun main(args: Array<String>) {

    val usage = """
        Usage: <functionName> 

        Where:
        functionName - the name of the Lambda function to invoke (for example, myLambda). 
       """
    if (args.size != 1) {
        println(usage)
        exitProcess(1)
    }

    val functionName = args[0]
    val lambdaClient = LambdaClient { region = "us-east-1" }
    invokeFunction(lambdaClient, functionName)
    lambdaClient.close()
}

// snippet-start:[lambda.kotlin.invoke.main]
suspend fun invokeFunction(lambdaClient: LambdaClient, functionNameVal:String) {

        val json = """{"Hello ":"Paris"}"""
        val byteArray = json.trimIndent().encodeToByteArray()

        try {

            val invokeRequest = InvokeRequest {
                functionName = functionNameVal
                payload = byteArray
            }

           val res = lambdaClient.invoke(invokeRequest)
           println(res)

        } catch (ex: LambdaException) {
            println(ex.message)
            lambdaClient.close()
            exitProcess(1)
        }
 }
// snippet-end:[lambda.kotlin.invoke.main]